### DISCUSSION TITLE
Are you planning to support webtransport based on tquic?

### DISCUSSION THREAD

**Discussion by zsoerenm** [Tencent/tquic#214] [Ideas（新功能的建议和讨论）]
**Describe the solution you'd like**
Webtransport based on tquic.

**Comment by glime-ai** [comment]
To implement WebTransport based on TQUIC, you would need to ensure that TQUIC supports the necessary protocols and features required by WebTransport. Based on the information provided in the context, TQUIC is an implementation of the IETF QUIC protocol, designed for high performance, low latency, and supports a wide range of features conforming to QUIC and HTTP/3 RFCs. It is cross-platform and written in Rust, offering APIs for Rust/C/C++.

For WebTransport over QUIC, you would need to ensure that TQUIC supports the specific ALPN (Application-Layer Protocol Negotiation) identifiers used by WebTransport. The ALPN is a TLS extension used for negotiating the application protocol in a TLS-encrypted connection, and WebTransport requires specific ALPN tokens to be recognized and supported by both the client and server.

Given that TQUIC supports QUIC and HTTP/3, and considering its flexible settings and detailed observability, it seems plausible that TQUIC could be adapted or configured to support WebTransport. However, the specific implementation details, such as how to configure SSL_CTX for non-HTTP ALPN parameters or handling application-layer data, would depend on TQUIC's current capabilities and API support.

To proceed with implementing WebTransport based on TQUIC, you should:

1. Review TQUIC's documentation and API reference to understand its current support for ALPN negotiation and how to handle application-layer data.
2. Check if TQUIC's latest version has explicit support for WebTransport or if there are any community contributions or discussions on this topic.
3. Consider reaching out to the TQUIC community or developers for guidance on implementing WebTransport, especially if specific features or configurations are not clearly documented.

Remember, the success of implementing WebTransport on TQUIC will largely depend on TQUIC's flexibility and the level of control it offers over the QUIC protocol's features, including ALPN negotiation and handling of application-layer data.

**Comment by iyangsj** [comment]
We're thinking about adding WebTransport support in future updates. But right now, we're focused on making sure our upcoming 1.0 version is stable and performs well. So, we won't add any big features until after it's out. After that, we plan to look into adding WebTransport support in our 2.0 or 3.0 versions.
